### DISCUSSION TITLE
Can I set collision ignore space near robot when I use nvblox?

### DISCUSSION THREAD

**Discussion by Sepirence** [NVlabs/curobo#148] [Software Q&A] [answered]
Hi, now I am controlling my ur5e robot with curobo. I succeeded to move real robot with your generated trajectory, and now I met a big huddle. I made nvblox world with depth camera, and generated mesh file and nvblox file. However, mesh contains robot meshes too, and I met self collision problem between robot in simulation and robot in mesh. I can delete meshes on mesh file, but I cannot edit easily to nvblox file. I am looking into your library, but is there a method that I can delete blox obstacles based on xyz coordinates?
As previous your replies always be my help, I will gladly wait for your answer.

**Comment by balakumar-s** [comment]
We currently do not have a way to segment out the robot. Is it possible for you to get the segmentation mask from sim?

We are working on adding robot segmentation within cuRobo but will take a few more weeks.

**Reply by Sepirence** [reply]
Maybe your work will be faster than me. I am just a fresh developer. But, if you give some inspection about implementation and specific work range, I will try to make it. 
My current approach is just add a cuboid on .yml file, and try to remove all mesh and all voxels in the cuboid, but it still has a problem that it still does not distinguish the voxels generated by real-time depth camera.

**Reply by balakumar-s** [reply]
One thing you can try is to add a box of 0's over the depth image where the robot exists, before sending to nvblox.

**Comment by balakumar-s** [comment, marked as answer]
Here is robot segmentation: https://curobo.org/source/advanced_examples/4_robot_segmentation.html

**Reply by DJIITBH** [reply]
I am using curobo segmentation but it segments the region of obstacle near robot arm and creates a mask on that also deleting the voxles of any object that comes near the robot, its really surprising to see such results I dont know why is it deleting the object's voxels within a region near the obstacle that too any where on the robot links. With this I am unable to run MPC on real robot as voxels near the robot region are not generated even if any object comes in its vicinity thus it never reacts. Please guide me on this. I can share the screenshots of rviz and real depth camera feed after segmentation for your reference.
